File: app/src/main/java/com/example/androidscript/util/BtnMaker.java
package com.example.androidscript.util;


import android.view.View;
import android.widget.Button;
import android.content.Intent;

import androidx.appcompat.app.AppCompatActivity;

import com.example.androidscript.Activities.Menu;


public final class BtnMaker {
    public static void Jump(int id, AppCompatActivity from, Class<?> to) {
        Button btn = from.findViewById(id);
        btn.setOnClickListener((v -> {
            Intent intent = new Intent(from, to);
            from.startActivity(intent);
            intent.putExtra("a", Menu.class);
        }));
    }

    public static void JumpWithMessage(int id, AppCompatActivity from, Class<?> to,String name,String message) {
        Button btn = from.findViewById(id);
        btn.setOnClickListener((v -> {
            Intent intent = new Intent(from, to);
            intent.putExtra(name, message);
            from.startActivity(intent);
        }));
    }

    public static void performIntentForResult(int id, AppCompatActivity from, Intent intent,int requestCode) {
        Button btn = from.findViewById(id);
        btn.setOnClickListener((v -> {
            from.startActivityForResult(intent, requestCode);
        }));
    }

    public static void registerOnClick(int id, AppCompatActivity from,View.OnClickListener listener) {
        Button btn = from.findViewById(id);
        btn.setOnClickListener(listener);
    }
}

File: app/src/main/java/com/example/androidscript/util/BtnMaker.kt
package com.example.androidscript.util

import androidx.appcompat.app.AppCompatActivity
import android.content.Intent
import android.view.*
import com.example.androidscript.activities.Menu

/** Helper class to simplify button construction.  */
object BtnMaker {
    fun startActivity(id: Int, from: AppCompatActivity, to: Class<*>) {
        from.findViewById<View>(id).setOnClickListener {
            val intent = Intent(from, to)
            from.startActivity(intent)
            intent.putExtra("a", Menu::class.java)
        }
    }

    fun startActivityWithMessage(
        id: Int,
        from: AppCompatActivity,
        to: Class<*>,
        name: String,
        message: String,
    ) {
        from.findViewById<View>(id).setOnClickListener {
            val intent = Intent(from, to)
            intent.putExtra(name, message)
            from.startActivity(intent)
        }
    }

    fun registerOnClick(id: Int, from: AppCompatActivity, listener: View.OnClickListener) {
        from.findViewById<View>(id).setOnClickListener(listener)
    }
}